app/global/MainModule.java

```diff
@@ -303,6 +308,33 @@ public ActorRef get() {
         private static final String INDEXER_ROLE = "host_indexer";
     }
 
+    private static final class RollupMetricsDiscoveryProvider implements Provider<ActorRef> {
+        @Inject
+        RollupMetricsDiscoveryProvider(final Injector injector, final ActorSystem system) {
+            _injector = injector;
+            _system = system;
+        }
+
+        @Override
+        public ActorRef get() {
+            final Cluster cluster = Cluster.get(_system);
+            if (cluster.selfRoles().contains(METRICS_DISCOVERY_ROLE)) {
+                return _system.actorOf(ClusterSingletonManager.props(
+                        GuiceActorCreator.props(_injector, MetricsDiscovery.class),
+                        PoisonPill.getInstance(),
+                        ClusterSingletonManagerSettings.create(_system).withRole(METRICS_DISCOVERY_ROLE)),
+                        "rollup-metrics-discovery"
+                );
+            }
+            return null;
+        }
+
+        private final Injector _injector;
+        private final ActorSystem _system;
+
+        private static final String METRICS_DISCOVERY_ROLE = "rollup_metrics_discovery";
```

Comment:
Rename `METRICS_DISCOVERY_ROLE` to `ROLLUP_METRICS_DISCOVERY_ROLE`; I can see this becoming a pattern.

Reply:
done